### PR TITLE
fix: added recommend_discharge as action

### PR DESCRIPTION
**Pull request by cp-Coder**
# Feature Request

## Proposed Changes

- Removed `recommended_discharge` field from `daily_round`
- Added `DISCHARGE_RECOMMENDED` as action

## Associated PR

- Required by coronasafe/care_fe#5347

## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [x] Linting Complete

@coronasafe/code-reviewers

### CODE REVIEW

**Review by siddnikh — APPROVED** [review]
LGTM

**Review by rithviknishad — APPROVED** [review]
@cp-Coder rebase the migrations

File: care/facility/migrations/0346_add_recommend_discharge_action.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+from django.db import migrations, models
+
+
+def update_recommend_discharge(apps, schema_editor):
+    patient_model = apps.get_model("facility", "PatientRegistration")
+    for patient in patient_model.objects.all():
+        if patient.last_consultation.last_daily_round.recommend_discharge:
+            patient.action = 90
+            patient.save()
```
Comment (sainak) [inline care/facility/migrations/0346_add_recommend_discharge_action.py:9]: use bulk update

File: care/facility/migrations/0346_add_recommend_discharge_action.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+from django.db import migrations, models
+
+
+def update_recommend_discharge(apps, schema_editor):
+    patient_model = apps.get_model("facility", "PatientRegistration")
+    for patient in patient_model.objects.all():
```
Comment (sainak) [inline care/facility/migrations/0346_add_recommend_discharge_action.py:6]: select related models as well

File: care/facility/models/daily_round.py [hunk 3]
```diff
@@ -122,7 +122,9 @@ class InsulinIntakeFrequencyType(enum.Enum):
     ]
 
     consultation = models.ForeignKey(
-        PatientConsultation, on_delete=models.PROTECT, related_name="daily_rounds"
+        PatientConsultation,
+        on_delete=models.PROTECT,
+        related_name="daily_rounds",
```
Comment (sainak) [inline care/facility/models/daily_round.py:127]: are these due to auto formatting, your formatter is not picking up the correct line length

**Review by sainak — CHANGES_REQUESTED** [review]
add tests

File: care/facility/models/daily_round.py (hunk 3 above)
Comment (cp-Coder) [reply care/facility/models/daily_round.py:127]: I am running pre-commit. All the files are formatted before I push the changes

File: care/facility/migrations/0346_add_recommend_discharge_action.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    patient_objs = []
+    for patient in patient_model.objects.all():
+        if (
+            patient.last_consultation
+            and patient.last_consultation.last_daily_round
+            and patient.last_consultation.last_daily_round.recommend_discharge
+        ):
+            patient.action = 90
+            patient_objs.append(patient)
+    patient_model.objects.bulk_update(patient_objs, ["action"])
```
Comment (rithviknishad) [inline care/facility/migrations/0346_add_recommend_discharge_action.py:15]: We could simply do the following instead right?
```suggestion
    PatientRegistration.objects.filter(last_consultation__last_daily_round__recommend_discharge=True).update(action=90)
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]
fix the tests CI failing

**Comment by cp-Coder** [comment]
I have added the tests for Log update. This PR is ready for review

**Comment by Ashesh3** [comment]
@cp-Coder Can you resolve the merge conflicts?

**Review by sainak — CHANGES_REQUESTED** [review]
there were some changes in tests so you need to create the required objects, refer other test

and revert changes in users/test_api and test_base

**Comment by nihal467** [comment]
Functionality LGTM @sainak @rithviknishad @Ashesh3 can you review it

**Review by rithviknishad — APPROVED** [review]

**Review by Ashesh3 — APPROVED** [review]

**Review by sainak — APPROVED** [review]

**Pull request merged**